database/sql/V048__create_settlement_rewards.sql

```diff
@@ -0,0 +1,29 @@
+-- Populated before settlement occurs on chain
+-- (after the submissions have been ranked)
+CREATE TABLE settlement_scores (
+  settlement_id bytea NOT NULL,
+  winning_score numeric(78,0) NOT NULL,
+  reference_score numeric(78,0) NOT NULL
+);
+
+-- Populated after block finalization via transactionReceipt.
+CREATE TABLE settlement_observations (
+  -- the appended settlement ID read from the call data
```

Comment:
If this table is populated after the auction is finished (for example lagging 64 blocks behind the latest block) that would be problematic, since we lose the `external_prices` vector once the auction is finished (and we also do not store the external prices anywhere right now I believe).
So, in order to avoid saving the whole `external_prices` vector, we would need to inspect the winning settlement before submission and save the subset of prices to db, and hope that the onchain transaction will be the same (just thinking out loud).

Reply:
But we can add the prices when we have them and create the link to here later...

Reply:
My current plan is to handle this as explained here: https://github.com/cowprotocol/services/pull/1166#discussion_r1098571919
Closing the thread.